lib/debug/ProfilingPlugin.js

```diff
@@ -266,9 +266,11 @@ const interceptTemplateInstancesFrom = (compilation, tracer) => {
 };
 
 const interceptAllHooksFor = (instance, tracer, logLabel) => {
-	Object.keys(instance.hooks).forEach(hookName => {
-		instance.hooks[hookName].intercept(makeInterceptorFor(logLabel, tracer)(hookName));
-	});
+	if(Object.hasOwnProperty("hooks", instance)) {
```

Comment:
This isn't doing what you are expecting.

```js
Object.hasOwnProperty("foo", {foo: 1}) // false
Object.prototype.hasOwnProperty.call({foo: 1}, "foo") // true
```

Please use `Reflect.has` or `in`. There is no compatibility issue since `@next` is targeting node 6+.

Reply:
no problem. Thanks!